docs/guides/fetch-data/02-read-asset-data.md

```diff
@@ -0,0 +1,1056 @@
+
+const InterfaceID = [
+  {
+    type: "function",
+    stateMutability: "view",
+    outputs: [
+      {
+        type: "bool",
+        name: "",
+        internalType: "bool",
+      },
+    ],
+    name: "supportsInterface",
+    inputs: [
+      {
+        type: "bytes4",
+        name: "interfaceId",
+        internalType: "bytes4",
+      },
+    ],
+  },
+];
+
+/*
+ * Check the interface of an
+ * asset's smart contract
+ *
+ * @returns: string of interface type
+ */
+async function checkAssetInterface(address) {
+
+  // Create instance of the contract which has to be queried
+  const AssetContract = new web3.eth.Contract(InterfaceID, address);
+
+  // Storing type of interface standard
+  let standard;
+
+  // Old version's hash of ERC725Y
+  const erc725YLegacy = "0x2bd57b73";
+  let isERC725YLegacy = false;
+
+  // Check if the contract is an old key-value store
+  try {
+    isERC725YLegacy = await AssetContract.methods
+      .supportsInterface(erc725YLegacy)
+      .call();
+  } catch (error) {
+    return console.log("Address could not be checked for lagacy interface");
+  }
+
+  // New version's hash of ERC725Y
+  const erc725Y = "0x5a988c0f";
+  let isERC725Y = false;
+
+  // Check if the contract is an old key-value store
+  try {
+    isERC725Y = await AssetContract.methods.supportsInterface(erc725Y).call();
+  } catch (error) {
+    return console.log("Address could not be checked for interface");
+  }
+
+  // Update standard variable
+  if (isERC725YLegacy) {
+    standard = "old";
+  }
+  if (isERC725Y) {
+    standard = "new";
+  }
+
+  return standard;
+}
+
+// Debug
+checkAssetInterface(sampleAssetAddress).then((standard) =>
+  console.log(standard)
+);
+```
+
+## Step 4 - Receive the encoded asset data
+
+Now we can safely call the data of the address. The [LSP4](../../standards/nft-2.0/LSP4-Digital-Asset-Metadata) data is saved in a 725Y key-value store, and we need to input the right key to fetch the associated value. There are multiple keys for different properties. To give a showcase, we will use the metadata key to receive the associated data.
+
+```javascript title="read_assets.js"
+...
+// ABI's
+const ERC725YLagacyMinimalInterface = [
+  {
+    inputs: [
+      {
+        internalType: 'bytes32',
+        name: '_keys',
+        type: 'bytes32',
+      },
+    ],
+    name: 'getData',
+    outputs: [
+      {
+        internalType: 'bytes',
+        name: 'values',
+        type: 'bytes',
+      },
+    ],
+    stateMutability: 'view',
+    type: 'function',
+  },
+];
+
+const ERC725MinimalInterface = [
+  {
+    inputs: [
+      {
+        internalType: 'bytes32[]',
+        name: '_keys',
+        type: 'bytes32[]',
+      },
+    ],
+    name: 'getData',
+    outputs: [
+      {
+        internalType: 'bytes[]',
+        name: 'values',
+        type: 'bytes[]',
+      },
+    ],
+    stateMutability: 'view',
+    type: 'function',
+  },
+];
+
+const LSP4schema = [
```

Comment:
There is this schema defined in _erc725.js_.
You can get it from the _erc725.js_ package directly.
https://github.com/ERC725Alliance/erc725.js/blob/develop/src/schemas/LSP4DigitalAsset.json

@Hugoo

Reply:
I'm not sure if the schema is exported - we should export it maybe

Reply:
I checked it, it is not exported to the npm package, only LSP1